src/validation/Validator.ts

```diff
@@ -47,6 +47,34 @@ export class Validator {
             return executor.stripEmptyErrors(validationErrors);
         });
     }
+
+    /**
+     * Performs validation of the given object based on decorators used in given object class.
+     */
+    restrictValidate(object: Object, options?: ValidatorOptions): Promise<void>;
+
+    /**
+     * Performs validation of the given object based on validation schema.
+     */
+    restrictValidate(schemaName: string, object: Object, options?: ValidatorOptions): Promise<void>;
+
+    /**
+     * Performs validation of the given object based on decorators or validation schema.
+     */
+    restrictValidate(objectOrSchemaName: Object|string, objectOrValidationOptions: Object|ValidationOptions, maybeValidatorOptions?: ValidatorOptions): Promise<void> {
```

Comment:
It's not that simple - when you have overload signatures, you can use only them, not the 3rd function `validate(objectOrSchemaName: Object|string, objectOrValidationOptions: Object|ValidationOptions, maybeValidatorOptions?: ValidatorOptions)`. I will move the implementation to the private core method.